submissions/AndriiZahumnyi/js-dom/script/index.js

```diff
@@ -0,0 +1,36 @@
+
+const hamburgerToggle = document.querySelector('#menu-toggle');
+
+const flexMenu = document.querySelector('.flex-menu');
+
+const arrAside = [".aside-html", ".aside-css", ".aside-bootstrap", ".aside-js", ".aside-jquery", ".aside-git"];
+
+const arrContent = [".html-section", ".css-section", ".bootstrap-section", ".js-section", ".jquery-section", ".git-section"];
+
+const asideMenuList = document.getElementById("aside-menu-list");
+
+const lengthOfArrAside = arrAside.length;
+
+function respondToTheClick(evt) {
```

Comment:
Sorry, it seems `Array.prototype.find` won't work there properly because you used different classes. Your decision works fine but it seems it has a lot overengineering staff.
I'd recommend you to use [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) instead of classes. Try to use classes for styling and the data-attributes for handling some logic staff.
You can add data-attributes with the same value to the menu items and to the content sections.
For example, you can add this attribute for the menu item: `data-menu` and for the content item: `data-content`. It helps you easily query necessary elements.
Also, instead of handling two classes on the section element you can handle one class. For example `'active'` class. You can make your sections hidden by default and use the active class on the visible (selected) section. So in this way you can simplify your decision: 
```javascript
function respondToTheClick({target}) {
  const matchedAttribute = target.dataset.menu;
  const activeSectionItem = document.querySelector('.section.active');
  activeSectionItem.classList.remove('active');
  const matchedSection = document.querySelector(`.section[data-content='${matchedAttribute}']`);
  matchedSection.classList.add('active');
}
```
Let me know if it's clear for you

Reply:
> Sorry, it seems `Array.prototype.find` won't work there properly because you used different classes. Your decision works fine but it seems it has a lot of overengineering staff.
> I'd recommend you to use [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) instead of classes. Try to use classes for styling and the data-attributes for handling some logic staff.
> You can add the data-attributes with the same value to the menu items and to the content sections.
> For example, you can add this attribute for the menu item: `data-menu` and for the content item: `data-content`. It helps you easily query the necessary elements.
> Also, instead of handling two classes on the section element, you can handle one class. For example `'active'` class. You can make your sections hidden by default and use the active class on the visible (selected) section. So in this way you can simplify your decision:
> 
> ```js
> function respondToTheClick({target}) {
>   const matchedAttribute = target.dataset.menu;
>   const activeSectionItem = document.querySelector('.section.active');
>   activeSectionItem.classList.remove('active');
>   const matchedSection = document.querySelector(`.section[data-content='${matchedAttribute}']`);
>   matchedSection.classList.add('active');
> }
> ```
> 
> Let me know if it's clear for you

Sorry, but my first variant with for loop? isn`t it OK? because first you said classe insted id, than you say data-attributes...
It seems to me that I will not done this stage, never!))

Reply:
Stop-stop! No worries) Everything is ok with your code and we can approve it without any doubts.
But to my mind, we can improve it a little bit and make it cleaner in order to prevent some misunderstandings in the next tasks which will be more complex than this one.
It seems we didn't get each other in our previous comments. So let's deal with it.
1. About the using classes instead of id I mean this case:
``` javascript
// NOT OK ❌
element.style.display = 'none';
// OK ✅
element.classList.add('hidden');
/* 
.hidden {
 display: none;
}
*/
```
2. Your decision with the loop is ok, but you did a lot of redundant operations there.
3. You can work with other tasks and don't wait for my approve or comments in this task.

Let me know if you have any objections. I'm ready to help you.